packages/hooks/use-throttle-render/index.ts

```diff
@@ -1,31 +1,53 @@
 import { onMounted, ref, watch } from 'vue'
+import { isNumber, isObject } from '@element-plus/utils'
 
+import type { SkeletonThrottle } from '@element-plus/components'
 import type { Ref } from 'vue'
 
-export const useThrottleRender = (loading: Ref<boolean>, throttle = 0) => {
+export const useThrottleRender = (
+  loading: Ref<boolean>,
+  throttle: SkeletonThrottle = 0
+) => {
   if (throttle === 0) return loading
-  const throttled = ref(false)
+  const initVal = isObject(throttle) ? throttle.initVal ?? false : false
+  const throttled = ref(initVal)
   let timeoutHandle: ReturnType<typeof setTimeout> | null = null
 
-  const dispatchThrottling = () => {
+  const dispatchThrottling = (timer: number) => {
     if (timeoutHandle) {
       clearTimeout(timeoutHandle)
     }
     timeoutHandle = setTimeout(() => {
       throttled.value = loading.value
-    }, throttle)
+    }, timer)
   }
-  onMounted(dispatchThrottling)
+
+  const dispatcher = (type: 'leading' | 'trailing') => {
+    if (type === 'leading') {
+      if (isNumber(throttle)) {
+        dispatchThrottling(throttle)
+      } else {
+        if (throttle?.leading && throttle.leading > 0) {
```

Comment:
The judgment here can be removed, otherwise, there will be problems when it is set to 0.

Reply:
Here `isNumber(throttle)` is for compatibility with the previous logic. `throttle?.leading && throttle.leading > 0` is here to determine that the `throttle` passed in by the user is an object, but the `leading` attribute is not passed. Here I think the judgment of `throttle.leading > 0` can also be removed.

Reply:
Yes, that's what I meant.

Reply:
I have thought about modifying it to this `const initVal = isObject(throttle) && Boolean(throttle.initVal)`